### ISSUE TITLE
`ERROR: column "public" of relation "buckets" does not exist (SQLSTATE 42703)` During `supabase db reset`

### ISSUE THREAD

**Issue by nbarrow-inspire-labs** [supabase/cli#1277]
**Describe the bug**
I was previously using v1.68.6 of the cli, and had no problem running migrations that included, among other things:
```plpgsql
-- supabase/migrations/20230420235920_buckets.sql
insert into storage.buckets (id, name, public)
  values
    ('public', 'public', 'TRUE'),
    ('private', 'private', 'FALSE');
```

I then upgraded to v1.75.3 and the following error began presenting itself:
```
Applying migration 20230420235920_buckets.sql...
ERROR: column "public" of relation "buckets" does not exist (SQLSTATE 42703)
At statement 0: insert into storage.buckets (id, name, public)              
  values                                                                    
    ('public', 'public', 'TRUE'),                                           
    ('private', 'private', 'FALSE')                                         
Try rerunning the command with --debug to troubleshoot the error.
```

I tried running the exact same query in an empty project, using v1.75.3, without any migrations, once `supabase db reset` finished (from the web UI's SQL Editor) and get no errors (`Success. 0 rows returned`). 

**To Reproduce**
1. Create a new supabase project without any migrations
```shell
cd ~ && mkdir supabase-bug-test
cd supabase-bug-test
supabase init && supabase start
```
2. Create a new migration: 
```shell
supabase migration new buckets
```
3. Add the following to the newly created migration:
```plpgsql
-- supabase/migrations/..._buckets.sql
insert into storage.buckets (id, name, public)
  values
    ('public', 'public', 'TRUE'),
    ('private', 'private', 'FALSE');
```
4. Attempt to reset the database (`supabase db reset`)

**Expected behavior**
The database should not be throwing the error (indeed, it worked all the way to, at least, v1.68.6 and broke somewhere between v1.75.3)

**Desktop (please complete the following information):**
 - OS: Mac OS 13.3 (22E252)
 - Version of CLI: 
   - v1.68.6 (no error)
   - v1.75.3 (error)

Edit: current workaround is downgrading back to 1.68.6:
```shell
brew tap-new supabase/local-tap                                      
brew extract --version=1.68.6 supabase/tap/supabase supabase/local-tap
brew install supabase@1.68.6
supabase -v
```

**Comment by sweatybridge** [comment]
Can you try `supabase stop --no-backup` before running `supabase start`? This flag cleans up any db volumes leftover from previous runs to ensure a clean start.

I can't seem to reproduce on a clean start.

**Comment by nbarrow-inspire-labs** [comment]
@sweatybridge I tried your suggestion, and running `supabase stop --no-backup` followed by `supabase start` manages to reset everything fine, however I am still getting the exact same issue when I try and run `supabase db reset`.

**Comment by nbarrow-inspire-labs** [comment]
Here's the output: 
```shell
nrb@nrb-mbp inspire-tms-frontend % supabase stop --no-backup
Stopped supabase local development setup.
nrb@nrb-mbp inspire-tms-frontend % supabase start
Applying migration 20230420161043_diff.sql...
Applying migration 20230420174112_diff.sql...
Applying migration 20230420184424_diff.sql...
Applying migration 20230420225315_diff.sql...
Applying migration 20230420230056_diff.sql...
Applying migration 20230420231240_diff.sql...
Applying migration 20230420231424_diff.sql...
Applying migration 20230420231755_diff.sql...
Applying migration 20230420235920_buckets.sql...
Applying migration 20230421000801_diff.sql...
Applying migration 20230421010505_diff.sql...
Applying migration 20230421015108_diff.sql...
Applying migration 20230421015218_diff.sql...
Applying migration 20230421015649_diff.sql...
Applying migration 20230421020201_diff.sql...
Applying migration 20230421020327_diff.sql...
Applying migration 20230421021224_diff.sql...
Applying migration 20230421022912_diff.sql...
Applying migration 20230421044427_diff.sql...
Applying migration 20230421181059_diff.sql...
Applying migration 20230423031435_diff.sql...
Applying migration 20230424160437_diff.sql...
Applying migration 20230424160753_diff.sql...
Applying migration 20230424162602_diff.sql...
Applying migration 20230424163352_diff.sql...
Applying migration 20230424203507_diff.sql...
Applying migration 20230425040001_diff.sql...
Applying migration 20230425040338_diff.sql...
Applying migration 20230425040741_diff.sql...
Applying migration 20230425041005_diff.sql...
Applying migration 20230425041836_diff.sql...
Applying migration 20230428031013_diff.sql...
Applying migration 20230501230442_diff.sql...
Applying migration 20230501230804_diff.sql...
Applying migration 20230501231041_diff.sql...
Applying migration 20230501231606_diff.sql...
Applying migration 20230501231948_diff.sql...
Applying migration 20230501232432_diff.sql...
Applying migration 20230501232717_diff.sql...
Applying migration 20230502000326_diff.sql...
Applying migration 20230503010236_diff.sql...
Applying migration 20230503023056_diff.sql...
Applying migration 20230503183826_diff.sql...
Applying migration 20230503184231_diff.sql...
Applying migration 20230503184738_diff.sql...
Applying migration 20230503184924_diff.sql...
Applying migration 20230503210636_diff.sql...
Applying migration 20230504031835_diff.sql...
Applying migration 20230504034722_diff.sql...
Applying migration 20230504151141_diff.sql...
Applying migration 20230504152212_diff.sql...
Applying migration 20230504152550_diff.sql...
Applying migration 20230504154129_diff.sql...
Applying migration 20230504173543_diff.sql...
Applying migration 20230505025826_diff.sql...
Applying migration 20230507010024_diff.sql...
Applying migration 20230507010811_diff.sql...
Applying migration 20230507011733_diff.sql...
Applying migration 20230507012148_diff.sql...
Applying migration 20230507012503_diff.sql...
Applying migration 20230508021949_diff.sql...
Applying migration 20230508022231_diff.sql...
Applying migration 20230508023214_diff.sql...
Applying migration 20230508023517_diff.sql...
Applying migration 20230508162346_diff.sql...
Applying migration 20230508162534_diff.sql...
Applying migration 20230508162800_diff.sql...
Applying migration 20230508165833_diff.sql...
Applying migration 20230508171021_diff.sql...
Applying migration 20230508220338_diff.sql...
Applying migration 20230508220712_diff.sql...
Applying migration 20230508221512_diff.sql...
Applying migration 20230508223935_diff.sql...
Applying migration 20230508230505_diff.sql...
Applying migration 20230509202321_diff.sql...
Applying migration 20230509203118_diff.sql...
Applying migration 20230509203432_diff.sql...
Applying migration 20230509204438_diff.sql...
Applying migration 20230511232238_diff.sql...
Applying migration 20230512014037_diff.sql...
Applying migration 20230512015357_diff.sql...
Applying migration 20230512020259_diff.sql...
Applying migration 20230515023038_diff.sql...
Applying migration 20230515221823_diff.sql...
Applying migration 20230515222533_diff.sql...
Applying migration 20230515223018_diff.sql...
Applying migration 20230515223622_diff.sql...
Applying migration 20230515224105_bulk_migrating_company_id_checks.sql...
Applying migration 20230515233429_diff.sql...
Applying migration 20230516000548_diff.sql...
Applying migration 20230516000712_diff.sql...
Applying migration 20230517050137_diff.sql...
Applying migration 20230518024632_diff.sql...
Applying migration 20230518024724_diff.sql...
Applying migration 20230518192318_diff.sql...
Applying migration 20230519172948_diff.sql...
Applying migration 20230526193217_company_id_unique_with_account_id.sql...
Applying migration 20230526193656_renaming_realtime_function.sql...
Applying migration 20230528180431_diff.sql...
Applying migration 20230528180548_diff.sql...
Applying migration 20230528180847_profile_picture_bucket.sql...
Applying migration 20230528182200_profile_picture_bucket_rls_policies.sql...
Applying migration 20230529013815_diff.sql...
Applying migration 20230529014029_diff.sql...
Applying migration 20230529014307_diff.sql...
Applying migration 20230530021845_diff.sql...
Applying migration 20230530041125_adding_current_company_id_to_accounts_with_protections.sql...
Applying migration 20230530042401_refining_account_level_protections.sql...
Applying migration 20230530043015_refining_account_level_protections.sql...
Applying migration 20230530043344_refining_account_level_protections.sql...
Applying migration 20230530153146_diff.sql...
Applying migration 20230531160824_delete_own_comments.sql...
Applying migration 20230531161242_view_profiles_by_user_company.sql...
Applying migration 20230531163817_fix_full_name_for_profiles.sql...
Applying migration 20230601001842_diff.sql...
Applying migration 20230606170234_diff.sql...
Applying migration 20230606170506_diff.sql...
Applying migration 20230608001156_diff.sql...
Applying migration 20230608001250_diff.sql...
Applying migration 20230608001429_protect_company_id_for_equipment.sql...
Applying migration 20230608002353_diff.sql...
Applying migration 20230608003515_finishing_equipment_table.sql...
Applying migration 20230608005510_diff.sql...
Applying migration 20230608150701_diff.sql...
Applying migration 20230608151045_diff.sql...
Applying migration 20230608171414_diff.sql...
Applying migration 20230608172040_diff.sql...
Applying migration 20230608183514_allow_auth_users_schema_access.sql...
Applying migration 20230609020218_beginning_public_issues.sql...
Applying migration 20230609021420_issues_company_id_protections.sql...
Applying migration 20230609021552_diff.sql...
Applying migration 20230609021812_diff.sql...
Applying migration 20230609031104_building_kanban.sql...
Applying migration 20230609031801_diff.sql...
Applying migration 20230609034612_diff.sql...
Applying migration 20230609035806_diff.sql...
Applying migration 20230609043139_diff.sql...
Applying migration 20230610040252_diff.sql...
Applying migration 20230610053206_diff.sql...
Applying migration 20230610055225_diff.sql...
Applying migration 20230611035549_diff.sql...
Applying migration 20230620025617_diff.sql...
Applying migration 20230621041533_diff.sql...
Applying migration 20230621154043_diff.sql...
Applying migration 20230622191520_diff.sql...
Applying migration 20230623152248_diff.sql...
Applying migration 20230623153101_protect_issues_company_id.sql...
Applying migration 20230623153234_diff.sql...
Applying migration 20230623153701_diff.sql...
Applying migration 20230627162130_diff.sql...
Applying migration 20230627184505_diff.sql...
Applying migration 20230627194932_diff.sql...
Applying migration 20230627202614_diff.sql...
Applying migration 20230627202707_diff.sql...
Applying migration 20230627204525_diff.sql...
Applying migration 20230628220435_diff.sql...
Applying migration 20230629052206_diff.sql...
Applying migration 20230629185356_diff.sql...
Applying migration 20230703232328_diff.sql...
Applying migration 20230704000252_removing_profile_pictures_bucket.sql...
Applying migration 20230704001743_public_profile_pictures_rls_policies.sql...
Applying migration 20230704002314_cleaning_up_old_profile_pictures_bucket_policies.sql...
Applying migration 20230704003053_profile_pictures_folder.sql...
Applying migration 20230704003913_diff.sql...
Applying migration 20230704004639_diff.sql...
Applying migration 20230704004651_create_public_profiles_picture_folder.sql...
Applying migration 20230704013509_allow_users_to_create_company_private_folder.sql...
Seeding data supabase/seed.sql...
Started supabase local development setup.

         API URL: http://localhost:54321
     GraphQL URL: http://localhost:54321/graphql/v1
          DB URL: postgresql://postgres:postgres@localhost:54322/postgres
      Studio URL: http://localhost:54323
    Inbucket URL: http://localhost:54324
      JWT secret: super-secret-jwt-token-with-at-least-32-characters-long
        anon key: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6ImFub24iLCJleHAiOjE5ODM4MTI5OTZ9.CRXP1A7WOeoJeXxjNni43kdQwgnWNReilDMblYTn_I0
service_role key: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZS1kZW1vIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImV4cCI6MTk4MzgxMjk5Nn0.EGIM96RAZx35lJzdJsyH-qQwv8Hdp7fsn3W0YpN81IU
nrb@nrb-mbp inspire-tms-frontend % supabase db reset
Resetting local database...
Restarting containers...
Applying migration 20230420161043_diff.sql...
Applying migration 20230420174112_diff.sql...
Applying migration 20230420184424_diff.sql...
Applying migration 20230420225315_diff.sql...
Applying migration 20230420230056_diff.sql...
Applying migration 20230420231240_diff.sql...
Applying migration 20230420231424_diff.sql...
Applying migration 20230420231755_diff.sql...
Applying migration 20230420235920_buckets.sql...
ERROR: column "public" of relation "buckets" does not exist (SQLSTATE 42703)
At statement 0: insert into storage.buckets (id, name, public)              
  values                                                                    
    ('public', 'public', 'TRUE'),                                           
    ('private', 'private', 'FALSE')                                         
Try rerunning the command with --debug to troubleshoot the error.
```

**Comment by sweatybridge** [comment]
Thank you for clarifying. I managed to reproduce and will release a fix soon.

**Comment by nbarrow-inspire-labs** [comment]
Thanks for the quick turn-around @sweatybridge !

**Comment by MichaelDeBoey** [comment]
@sweatybridge It seems tlike there's a regression and #1278 got reverted in v2.x for some reason as I currently have the exact same problem as described above with the latest version of the CLI

Apparently @AbdullahAhmadAAK had the same problem a few months ago with supabase/supabase#29804, but that one got closed by @Hallidayo due to inactivity

I would be happy to contribute a fix, but I'm not familiar with Go as a language unfortunately, so I'm hoping this one can get resolved by somebody from the supabase team 🙏

**Comment by andrew-wang-polar** [comment]
@MichaelDeBoey I had the same issue. Upgrading the supabase cli from 2.31.8 to 2.45.5 resolved it

**Comment by grant** [comment]
I'm seeing the same issue today:

```sh
ERROR: column "public" of relation "buckets" does not exist (SQLSTATE 42703)
At statement: 0
-- Create the migration-blobs bucket
insert into storage.buckets
  (id, name, public, file_size_limit)
             ^
Try rerunning the command with --debug to troubleshoot the error.
```

❯ which supabase
/opt/homebrew/bin/supabase
❯ supabase --version
2.45.5

**Comment by benschac** [comment]
SAME

ERROR: column "public" of relation "buckets" does not exist (SQLSTATE 42703)       
At statement: 0                                                                    
-- Create a new bucket called 'avatars' with appropriate settings                  
insert into storage.buckets (id, name, public, file_size_limit, allowed_mime_types)

which supabase
/opt/homebrew/bin/supabase

supabase --version
2.45.5

**Comment by maxtuzz** [comment]
Yep me too on 2.45.5

**Comment by RichardHruby** [comment]
We are also having problems on 2.45.5 as well.

`ERROR: column "public" of relation "buckets" does not exist (SQLSTATE 42703)`

**Comment by jg33** [comment]
Hi- I think I'm experiencing something similar: 

`ERROR: function storage.enforce_bucket_name_length() does not exist (SQLSTATE 42883)
At statement: 221
CREATE TRIGGER enforce_bucket_name_length_trigger BEFORE INSERT OR UPDATE OF name ON storage.buckets FOR EACH ROW EXECUTE FUNCTION storage.enforce_bucket_name_length()`

also 2.45.5

**Comment by r-bt** [comment]
Also experiencing this issue on version 2.45.5 of the CLI

**Comment by maxtuzz** [comment]
Has anyone got a workaround?

**Comment by kaiba42** [comment]
hey yes, main branch of the cli repo has the fix

**Comment by kaiba42** [comment]
you can compile the binary yourself with `go build -o supabase .`, and just use it with any sub command e.g. `./supabase start`

**Comment by kaiba42** [comment]
hopefully devs publish the latest soon

**Comment by maxtuzz** [comment]
Thanks for digging @kaiba42, I did that and set up a `supabase-custom` alias. All working and I'm unblocked

**Comment by williamneves** [comment]
2.45.5

same with me

**Comment by grant** [comment]
I believe `supabase@beta` fixes this.

**Comment by ianoxwell** [comment]
Confirming that `npx supabase@beta start` does fix issue with 'public' in storage buckets
Also note that you must continue to use `npx supabase@beta db reset`

Further to help understanding of the bug - if you remove any reference to public, file_size_limit, allowed_mime_types from migration files you can manually run sql in the local editor after start

```sql
UPDATE storage.buckets 
SET public = true
where id = 'banners'
   OR id = 'cards'
   OR id = 'logos'
   OR id = 'products';
```

Link to another bug - https://github.com/supabase/cli/issues/4191
thank you to @sweatybridge for his response in ticket 4191

**Comment by jdgamble555** [comment]
I'm having this issue too.  Why is this closed?

**Comment by vickkhera** [comment]
> I'm having this issue too. Why is this closed?

CLI version 2.48.3 seems to fix it (that's the latest version today, don't know if an earlier version also fixes it).

**Comment by benschac** [comment]
hi, had to pull a new image. same issue again.

**Comment by benschac** [comment]
`npx supabase@beta db reset` fixes the issue.

more context

```
v1.26.4: Pulling from supabase/storage-api
```
is broken

via beta
```
v1.28.1: Pulling from supabase/storage-api
```

works

**Comment by efitzkiwi** [comment]
Can't this be prevented in future releases with a simple integration test? Super frustrating

**Comment by sweatybridge** [comment]
Yes, I've added new safeguards in the latest CLI release to sync both the storage image version and the storage migration version when linking a project. This should be more robust than relying on upgrading to latest cli.

**Comment by RichardHruby** [comment]
We are seeing this issue re-surface. Currently on `v.2.47.2`

**Comment by RichardHruby** [comment]
Deleting all docker images and updating to latest `v2.58.5` doesn't fix the issue.

**Comment by MichaelDeBoey** [comment]
@sweatybridge Seems like there's still something wrong as I have the same problem again

**Comment by MichaelDeBoey** [comment]
@sweatybridge You seem to have fixed the problem in #4429, which is present in `v2.58.8`, but that's come to `latest` tag (yet) and is marked as pre-release on GitHub as well

**Comment by RichardHruby** [comment]
I found that the supabase CLI generates a `.temp` directory when executing `supabase link`.

If I delete this directory and I don't explicitly call `supabase link` before calling `supabase db reset` then the reset completes successfully on version `v.2.47.2`. 

If I do call `supabase link` again, the temp folder re-generates with a number of files including `storage-version` which I suspect to be related to this issue.
